src/main/java/bio/terra/stairway/Flight.java

```diff
@@ -0,0 +1,177 @@
+package bio.terra.stairway;
+
+import bio.terra.stairway.exception.StairwayExecutionException;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+/**
+ * Manage the atomic execution of a series of Steps
+ * This base class has the mechanisms for executing the series of steps.
+ *
+ *  In order for the flight to be re-created on recovery, the construction and
+ *  configuration have to result in the same flight given the same input.
+ *
+ *  ISSUE: handling InterruptedException - is there anything that the Flight level should do
+ *  for handling this?
+ */
+public class Flight implements Callable<FlightResult> {
+    static class StepRetry {
+        private Step step;
+        private RetryRule retryRule;
+
+        StepRetry(Step step, RetryRule retryRule) {
+            this.step = step;
+            this.retryRule = retryRule;
+        }
+    }
+
+    private List<StepRetry> steps;
+    private Database database;
+    private FlightContext flightContext;
+
+    public Flight(SafeHashMap inputParameters) {
+        flightContext = new FlightContext(inputParameters).flightClassName(this.getClass().getName());
+        steps = new LinkedList<>();
+    }
+
+    public FlightContext context() {
+        return flightContext;
+    }
+
+    void setFlightContext(FlightContext flightContext) {
+        this.flightContext = flightContext;
```

Comment:
Not doing fluent-style here? Seems useful to be consistent

Reply:
We agreed in our meeting that we would use fluent for builder model and set for classes. I think the code now conforms to that choice.